### DISCUSSION TITLE
Video Upload

### DISCUSSION THREAD

**Discussion by dakom** [ji-devs/ji-cloud#1843] [Ideas]
Best would be to encode to some adaptive bitrate tech, like DASH or HLS. 

When we originally discussed this idea, it was very far out of scope, since our encoding was being done on the API server and that wold have been far too heavy.

Now that we have transcoding on its own server, this is actually doable, without requiring the expense of a third-party service like Bitmovin. Here's an example of some ffmpeg commands: https://streaminglearningcenter.com/blogs/an-ffmpeg-script-to-render-and-package-a-complete-hls-presentation.html

Since transcoding takes time, it would probably also have to be put in a Queue (and Google has some background services to manage that at scale)

The alternative would use a third-party service like Bitmovin.

_fwiw, the legacy player needed to support manual video, but TT's legacy player only plays the single uploaded file, and we did not improve on that... so it's technically working, but only truly in a "legacy support" sense, not ideal_

**Comment by corinnewo** [comment]
We need to look into http://cloudinary.com/ as well.
